From e38ef0c67e036fa96106846de7e0e80bb9039c6e Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Tue, 24 Jan 2012 13:04:50 +0000
Subject: Fix window MRU order.

Commit 0df941c26aef7bccf8c8530b41bdc7e9af0aaa9e introduced giving
selected windows focus in order to allow keyboard shortcuts being run on
them. This had the unwanted side effect of messing up the MRU order. Fix
that by not moving focus, but just temporarily changing activeWindow.

diff -Nur a/src/staticswitcher/staticswitcher.c b/src/staticswitcher/staticswitcher.c
--- a/src/staticswitcher/staticswitcher.c	2011-04-21 12:20:09.000000000 +0200
+++ b/src/staticswitcher/staticswitcher.c	2014-08-01 18:41:50.100033254 +0200
@@ -474,7 +474,7 @@
 	}
 
 	ss->selectedWindow = w;
-	moveInputFocusToWindow (w);
+	s->display->activeWindow = w->id;
 
 	if (old != w)
 	{
@@ -760,20 +760,11 @@
 	    }
 
 	    ss->switching = FALSE;
+	    d->activeWindow = sd->lastActiveWindow;
 
-	    if (state & CompActionStateCancel)
-	    {
-		ss->selectedWindow = NULL;
-		if (d->activeWindow != sd->lastActiveWindow)
-		{
-		    w = findWindowAtDisplay (d, sd->lastActiveWindow);
-		    if (w)
-			moveInputFocusToWindow (w);
-		}
-	    }
-
-	    if (state && ss->selectedWindow && !ss->selectedWindow->destroyed)
-		sendWindowActivationRequest (s, ss->selectedWindow->id);
+	    if (state && !(state & CompActionStateCancel))
+		if (ss->selectedWindow && !ss->selectedWindow->destroyed)
+		    sendWindowActivationRequest (s, ss->selectedWindow->id);
 
 	    removeScreenGrab (s, ss->grabIndex, 0);
 	    ss->grabIndex = 0;
